Add missing `git` icons

danielbayley opened the pull request:
* Added `git-commit-vertical` (and subsequently renamed `git-commit`[`-horizontal`])
* Added `git-pull-request-create`
* Refined `git-fork` to better match [`network`](https://lucide.dev/icons/network)
* Refined `git-pull-request-closed` to better match other `*-x` icons (as most `x`es are 5x5 px, not 6x6)
* Added `git-`[`pull-request-`[`create`/`compare`]`-arrow`/`s` variants, only making them slightly wider for better consistency than just scaling them up on both axes.
* Added `git-graph` icon (maybe should be `git-commit-graph`?) like in VS Code…

<img width="347" alt="Untitled" src="https://github.com/lucide-icons/lucide/assets/7797479/b1fcec1a-1b74-4a81-bcdc-debc1cfcaae3">
<br/>
<br/>

Full preview of changes in https://github.com/lucide-icons/lucide/pull/995#issuecomment-1509285280.

karsa-mistmere commented:
I'm afraid the bookmark and the tiny arrows do not have sufficient padding (a minimum of 2px spacing would be required by our guidelines).

karsa-mistmere commented:
These latest icons should probably use some sort of name along the lines of `inbox-{...}`, but they also have some guideline violations I'm afraid:
![image](https://user-images.githubusercontent.com/17746067/228624630-e80a7513-e4e4-4eee-92fc-7dc30de144b0.png)

danielbayley commented:
> I'm afraid the bookmark and the tiny arrows do not have sufficient padding (a minimum of 2px spacing would be required by our guidelines).

Tiny arrows, which do you mean? One of them was misaligned by mistake, fixed now…

karsa-mistmere commented:
> Tiny arrows, which do you mean? One of them was misaligned by mistake, fixed now…

E.g. the spacing marked with yellow here: 
![image](https://user-images.githubusercontent.com/17746067/228629796-cdb33839-a3e3-43a0-ab69-9710da317234.png)

danielbayley commented:
Other alternative would be:

<img width="384" alt="Artboard 1@30x" src="https://user-images.githubusercontent.com/7797479/228640247-840d731b-ee02-4227-9fee-059258993fb2.png">

danielbayley commented:
Or, maybe:

![inbox-2](https://user-images.githubusercontent.com/7797479/228643874-9ace4e1d-63d0-4a5b-95d3-fa18ddf6d733.png)

danielbayley commented:
Or, maybe:

![inbox-2](https://user-images.githubusercontent.com/7797479/228644192-b0a3a7a1-29b4-4875-aeac-c3d393e94f03.png)

danielbayley commented:
Or, maybe:

<img width="384" alt="inbox-2" src="https://user-images.githubusercontent.com/7797479/228644371-61b9eae5-3f66-4065-b3da-5333b1a58d10.png">

danielbayley commented:
@karsa-mistmere Fixed the repo icons…

<img width="384" alt="repos" src="https://user-images.githubusercontent.com/7797479/228684387-6312831a-8cd8-4a73-b1bf-70e8eb1bd988.png">

ericfennis reviewed (commented):
Thanks for these new icons. So next time try to split up your icons with multiple PRs grouped visually. This makes reviewing icon PRs a lot easier. 

For this PR:
I agree with @karsa-mistmere the arrows are a bit too close to the line. If you preview the icon on 100% it looks a bit messy. Not sure how you can fix this. Will try to help you with that one.

For the `git-repo-*` icons: 
Would be nice if the key and the lock are visually the same as the ones used in the `files` icons. And I'm not sure about the naming. The icons can be used for other things as `git-repos` as well so maybe we can give them more generic names.

And the `git-stash` icons should be named `inbox-in` and `inbox-out`. You may better add them to your other PR #998, I think they belong there.

danielbayley commented:
>  So next time try to split up your icons with multiple PRs grouped visually. This makes reviewing icon PRs a lot easier.

Yeah, I intended to… Just bundled all the `git` related ones together for now, I’ll break them down more now there is a better consensus…

I’ll move the repo icons to a separate PR, as I am happy with the versions I’ve tweaked offline… and I’ll leave this one open for the `git` command icons, as the arrows on those are trickier to solve.

> I'm not sure about the naming. The icons can be used for other things as `git-repos` as well so maybe we can give them more generic names.

Something like `book-marked`?

> And the `git-stash` icons should be named `inbox-in` and `inbox-out`. You may better add them to your other PR https://github.com/lucide-icons/lucide/pull/998, I think they belong there.

Agreed…

ericfennis commented on icons/git-pull-request-2.svg:
```diff
@@ -0,0 +1,17 @@
+<svg
```
I'm not pleased with the current size of the arrow from this one. 

I tried to give one a shot. But It's hard. I came up with two alternatives, but they have some tradeoffs..

ericfennis replied:
Forgot to post the screenshot. Let me know what you think
<img width="809" alt="Screenshot 2023-04-10 at 11 28 12" src="https://user-images.githubusercontent.com/11825403/231103062-970affa4-8a86-4271-946f-f7d64fd89415.png">

danielbayley replied:
@ericfennis I think alt 1…

ericfennis replied:
What do you think @karsa-mistmere ?

karsa-mistmere replied:
Honestly, I guess I'm mostly in favour of keeping just the ones with no arrows. 🤷‍♂️

To me, both alternatives feel forced; Alt 1 will make our git icons inconsistent and Alt 2 doesn't quite stick to the guidelines. 🙈

danielbayley replied:
> Alt 1 will make our git icons inconsistent

Although they could easily be made consistent… Also visually centred and proportionally more square:

<img width="418" alt="Untitled" src="https://user-images.githubusercontent.com/7797479/233218329-33c3b0b1-eaea-486c-92ba-a8c5d2bbb839.png">

ericfennis commented:
Still not sure about breaking the consistency with other git accounts. But the arrows are better now.
I keep this one open, I want to know what others think of this.

danielbayley commented:
> Still not sure about breaking the consistency with other git accounts.

I don’t think they should be inconsistent either, but no reason why they have to be… For reference:

<img width="420" alt="Untitled" src="https://user-images.githubusercontent.com/7797479/235292173-6d90ca6f-0db1-47a1-bbe9-319787a20391.png">

Top row are additions to the existing set, with the same proportions. Row 2 is the existing set. Row 3 are adapted to keep consistency with row 4, which are the arrow variations.

Thoughts? @ericfennis @karsa-mistmere @jguddas

danielbayley commented:
> can you provide these in a Figma file first so that I can play around with them a bit? :)

@karsa-mistmere https://figma.com/file/LjycnaUj6vyISowz39qNfX/lucide%2Fgit-icons?node-id=0%3A1&t=uGJ176Kfok9Y3qfg-1

danielbayley commented:
See latest update… @karsa-mistmere @ericfennis @jguddas

ericfennis reviewed (commented):
Ok In general some looks not that bad taking a bit more space.
I agree with @karsa-mistmere the `git-pull-request-closed` and `git-pull-request-draft` looking not really balanced. 

For me, I prefer to keep the old designs for `git-compare` and `git-pull-request` and add your new designs as alternative icons. So under a new name or add this as as second variant.

danielbayley commented:
> For me, I prefer to keep the old designs for `git-compare` and `git-pull-request` and add your new designs as alternative icons. So under a new name or add this as as second variant.

Ok, so I `revert`ed scaling them all up, and instead implemented the following summary of changes…

* Added `git-commit-vertical` (and subsequently renamed `git-commit`[`-horizontal`])
* Added `git-pull-request-create`
* Refined `git-fork` to better match [`network`](https://lucide.dev/icons/network)
* Refined `git-pull-request-closed` to better match other `*-x` icons (as most `x`es are 5x5 px, not 6x6)
* Added `git-`[`pull-request-`[`create`/`compare`]`-arrow`/`s` variants, only making them slightly wider for better consistency than just scaling them up on both axes.
* Added `git-graph` icon as outlined [above](https://github.com/lucide-icons/lucide/pull/995#issue-1644855557)

@ericfennis See latest https://github.com/lucide-icons/lucide/pull/995#issuecomment-1509285280.

cc: @karsa-mistmere @jguddas

danielbayley commented on icons/git-pull-request-closed.svg:
```diff
@@ -12,7 +12,7 @@
   <path d="M18 11.5V15" />
-  <path d="m21 3-6 6" />
-  <path d="m21 9-6-6" />
+  <path d="m20.5 3.5-5 5" />
```
@ericfennis Fixed.
